### PR TITLE
Crud employee  ilai azulay

### PR DESCRIPTION
**Pull request by ilaiazulay**
Still have to notify the user if edit or delete use succeed

### CODE REVIEW

File: .idea/.gitignore [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Default ignored files
```
Comment (DaveAvigad) [inline .idea/.gitignore:1]: no need to push .idea library (not necessary)

File: controllers/CRUDcontroller.js [hunk 2]
```diff
@@ -0,0 +1,110 @@
+const fs = require('fs');
```
Comment (DaveAvigad) [inline controllers/CRUDcontroller.js:1]: `CRUDcontroller.js` is not a proper file name
please use the common pattern  [context].[type].[ext]
for example:
`user.repository.js`
`login.controller.js`
`account.model.js`

File: controllers/CRUDcontroller.js [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+const handleDisplay = (jsonfile) => {
+    console.log(jsonfile)
+    return JSON.stringify(jsonfile);
```
Comment (DaveAvigad) [inline controllers/CRUDcontroller.js:5]: 🎉

File: controllers/CRUDcontroller.js [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+};
+
+
+const handleAdd = (jsonfile, queryObj) => {
```
Comment (DaveAvigad) [inline controllers/CRUDcontroller.js:9]: `handleAdd` not clear enough
please try a better declaration 😉

File: controllers/CRUDcontroller.js [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+    try {
+        var data = {};
```
Comment (DaveAvigad) [inline controllers/CRUDcontroller.js:11]: var not allowed. please use ES6^

File: controllers/CRUDcontroller.js [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+
+        emailInput=queryObj.email;
```
Comment (DaveAvigad) [inline controllers/CRUDcontroller.js:13]: `emailInput`
`passwordInput`
`nameInput`
`vacationDaysLeftInput`
where is the definition of these variables?

File: controllers/CRUDcontroller.js [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+        passwordInput=queryObj.password;
+        nameInput=queryObj.name;
+        vacationDaysLeftInput=queryObj.vacationDaysLeft;
+        var obj = {
```
Comment (DaveAvigad) [inline controllers/CRUDcontroller.js:17]: no `var`
change `obj`  variable name

File: controllers/CRUDcontroller.js [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+            email: emailInput,
+            password: passwordInput,
+            name: nameInput,
+            role: "employee",
+            vacationDaysLeft: vacationDaysLeftInput
+        };
+         jsonfile["users"].push(obj);
+         console.log(jsonfile);
+
+        fs.writeFileSync('./users.json', JSON.stringify(jsonfile, null, 2), function(err) {
```
Comment (DaveAvigad) [inline controllers/CRUDcontroller.js:27]: why use sync method? `writeFileSync`
you block the event loop. big no-no.

File: controllers/CRUDcontroller.js [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+        );
+    } catch (error) {
+        throw error;
+    }
+};
+
+const handleDelete = (jsonfile, queryObj) => {
+    try {
+        emailInput=queryObj.email;
+        var jsonArray = jsonfile["users"];
```
Comment (DaveAvigad) [inline controllers/CRUDcontroller.js:40]: replace all var declarations into Es6^

File: controllers/CRUDcontroller.js [hunk 10]
```diff
@@ -0,0 +1,110 @@
...
+        var del=[];
```
Comment (DaveAvigad) [inline controllers/CRUDcontroller.js:41]: please use a linter to prettify your code.
@AtomicManic this should be in the project root, for all files.
https://dev.to/drsimplegraffiti/eslint-configuration-for-node-project-275l

File: controllers/CRUDcontroller.js [hunk 11]
```diff
@@ -0,0 +1,110 @@
...
+        var foundFlag = 0;
+        for(var k in jsonArray)
```
Comment (DaveAvigad) [inline controllers/CRUDcontroller.js:43]: k 🤔

File: controllers/CRUDcontroller.js [hunk 12]
```diff
@@ -0,0 +1,110 @@
...
+        {
+            console.log(jsonArray[k].email)
+            if(jsonArray[k].email === emailInput )
+            {
+                foundFlag = 1;
```
Comment (DaveAvigad) [inline controllers/CRUDcontroller.js:48]: `foundFlag` --> found is enough

File: controllers/CRUDcontroller.js [hunk 13]
```diff
@@ -0,0 +1,110 @@
...
+                console.log("found");
```
Comment (DaveAvigad) [inline controllers/CRUDcontroller.js:49]: `console.log("found")` is redundant

File: controllers/CRUDcontroller.js [hunk 14]
```diff
@@ -0,0 +1,110 @@
...
+                console.log(jsonfile);
+            }
+        }
+        if(foundFlag == 0)
+            console.log("user not found");
+        else
+        {
+            fs.writeFileSync('./users.json', JSON.stringify(jsonfile, null, 2), function(err) {
+                    if (err) throw err;
+                    console.log('complete');
```
Comment (DaveAvigad) [inline controllers/CRUDcontroller.js:61]: `console.log('complete')` is redundant here

File: controllers/CRUDcontroller.js [hunk 15]
```diff
@@ -0,0 +1,110 @@
...
+                }
+            );
+        }
+    } catch (error) {
+        throw error;
+    }
+};
+
+const handleEdit = (jsonfile, queryObj) => {
```
Comment (DaveAvigad) [inline controllers/CRUDcontroller.js:70]: handle edit what? 
please give it a name that explains this method.
such as handleEmployeeRequest / editVacationHandler

File: controllers/CRUDcontroller.js [hunk 16]
```diff
@@ -0,0 +1,110 @@
...
+                jsonArray[k].password = passwordInput;
+                jsonfile["users"] = jsonArray
+                console.log(jsonfile);
+            }
+        }
+        if(foundFlag == 0)
+            console.log("user not found");
+        else
+        {
+            fs.writeFileSync('./users.json', JSON.stringify(jsonfile, null, 2), function(err) {
```
Comment (DaveAvigad) [inline controllers/CRUDcontroller.js:94]: replace `function` please use Es6^

File: controllers/loginController.js [hunk 17]
```diff
@@ -0,0 +1,15 @@
+
+const fs = require("fs");
+
+const handleLogin = (queryObj) => {
+    const rawData = JSON.parse(fs.readFileSync("users.json"));
```
Comment (DaveAvigad) [inline controllers/loginController.js:5]: replace `readFileSync` into async function

File: package.json [hunk 18]
```diff
@@ -5,12 +5,19 @@
...
   },
   "author": "",
   "license": "ISC",
   "devDependencies": {
+    "@types/node": "^18.11.9",
     "nodemon": "^2.0.20"
+  },
+  "dependencies": {
+    "cors": "^2.8.5",
+    "express": "^4.18.2",
```
Comment (DaveAvigad) [inline package.json:19]: no using express at this phase

**Review by DaveAvigad — COMMENTED** [review]
nice job, 
you have few things to fix.
please pay attention to variable names, modular code and clean code.

File: controllers/CRUDcontroller.js (hunk 8 above)
Comment (AtomicManic) [reply controllers/CRUDcontroller.js:27]: Look at my code -I was able to fix this as well with fs promise.

File: controllers/CRUDcontroller.js (hunk 12 above)
Comment (AtomicManic) [reply controllers/CRUDcontroller.js:48]: Could also be isFound

File: controllers/loginController.js [hunk 19]
```diff
@@ -0,0 +1,15 @@
...
+    const users = rawData.users;
+    for (const user of users) {
+        if (queryObj.email === user.email) {
+            return JSON.stringify(user);
+        }
+    }
+    return JSON.stringify(JSON.parse("{}"));
```
[the remark this answers is not in the document]
Comment (AtomicManic) [reply controllers/loginController.js:12]: If it is already a string you don't need to parse and stringify it

File: index.js [hunk 20]
```diff
@@ -0,0 +1,99 @@
+const http = require("http");
+const url = require("url");
+const usersJson = require("./users.json");
+const handleCRUD = require("./controllers/CRUDcontroller");
+const {json} = require("stream/consumers");
+const {response} = require("express");
+
+const port = process.env.PORT | 8081;
+
+const server = http
```
[the remark this answers is not in the document]
Comment (AtomicManic) [reply index.js:10]: Look at my code, I already did that

**Pull request merged**